plugins/optimization-detective/class-od-url-metrics-post-type.php

```diff
@@ -0,0 +1,329 @@
+<?php
+/**
+ * Optimization Detective: OD_URL_Metrics_Post_Type class
+ *
+ * @package optimization-detective
+ * @since 0.1.0
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * URL Metrics Post Type.
+ *
+ * @since 0.1.0
+ * @access private
+ */
+class OD_URL_Metrics_Post_Type {
+
+	/**
+	 * Post type slug.
+	 *
+	 * @var string
+	 */
+	const SLUG = 'od_url_metrics';
+
+	/**
+	 * Event name (hook) for garbage collection of stale URL Metrics posts.
+	 *
+	 * @var string
+	 */
+	const GC_CRON_EVENT_NAME = 'od_url_metrics_gc';
+
+	/**
+	 * Recurrence for garbage collection of stale URL Metrics posts.
+	 *
+	 * @var string
+	 */
+	const GC_CRON_RECURRENCE = 'daily';
+
+	/**
+	 * Registers post type for URL metrics storage.
+	 *
+	 * This the configuration for this post type is similar to the oembed_cache in core.
+	 *
+	 * @since 0.1.0
+	 */
+	public static function register() {
```

Comment:
Using static methods should usually be reserved for stateless functionality, however most of the methods in this class modify state (either by what's registered in WordPress or the database). Static methods are difficult and sometimes impossible to unit test if they modify state.

To be fair, in the specific cases here I don't see a strong reason to make them regular instance methods. But we can't change it later once this is part of an API, so I think it would still be a more future-proof choice to avoid static methods.

We can add a static `instance()` method or similar to make it trivial to get the "main instance" of the class. That doesn't make the code more complicated, but enhances testability for the future.

Reply:
Humm, but the class is stateless. There aren't any instances variables. A namespace would essentially work the same here as a class.

Reply:
Also, this class is marked as being `@access private` (although `@internal` maybe would be better). So the API is advertising as not being public for others to use.

Reply:
The class isn't really stateless even though it looks like it. It calls global functions that modify state in WordPress. I think the only method in the class that is actually stateless is the `parse_post_content()` method which returns URL metrics from a given post.

The other methods either change state of WordPress or at least rely on persistence layers outside of the class.

Reply:
I meant stateless in the context of the class. So there's nothing here about this being `static` that impacts testability.

Reply:
Yeah that's fair. Let's keep as is for now, given the class isn't part of any external API (via the `@access private`).